src/components/Groups/GroupAdd.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <div>
+    <h1>Add new group</h1>
+    <b-card-group deck>
+      <b-card bg-variant="light">
+
+        <b-alert
+          :show="!!addGroupErrorMessage"
+          variant="danger"
+          dismissible
+          @dismissed="addGroupErrorMessage=undefined">
+          {{ addGroupErrorMessage }}
+        </b-alert>
+
+        <b-form @submit="addNewGroup">
+
+          <b-form-group
+            :state="groupNameState"
+            label="Group name"
+            label-for="input-groupname"
+            horizontal
+          >
+            <b-form-input
+              id="input-groupname"
+              :state="groupNameState"
+              v-model="addGroupName"
+              type="text"
+              autofocus
+            />
+          </b-form-group>
+
+          <b-form-group>
+            <b-button
+              type="submit"
+              variant="primary">
+              Add group
+            </b-button>
+          </b-form-group>
+        </b-form>
+
+      </b-card>
+    </b-card-group>
+  </div>
+</template>
+
+<script>
+const groupNameMinLength = 3;
+
+export default {
+  name: 'GroupAdd',
+  data() {
+    return {
+      addGroupErrorMessage: null,
+      addGroupName: ""
+    };
+  },
+  computed: {
+    groupNameState () {
+      return this.addGroupName.length >= groupNameMinLength;
```

Comment:
Perhaps return null if length is zero, to avoid a red box when the page is initially viewed (as the user won't always be looking to add a new group)

![redbox](https://user-images.githubusercontent.com/24260476/43164213-b09dd2da-8fe4-11e8-9d45-97e05714627a.png)
I did the following on the register page, but there might be a more elegant solution.
```
if (this.password.length === 0) {
        // Empty field
        return null;
      } else {
        return this.password.length > passwordLength;
      }
```